File: app/src/main/java/com/damhoe/skatscores/ApplicationModule.java
package com.damhoe.skatscores;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {

   private final Application mApplication;

   @Singleton
   protected ApplicationModule(Application app) {
      mApplication = app;
   }

   @Provides
   @ApplicationContext
   Context provideAppContext() {
      return mApplication;
   }

   @Provides
   Context provideContext(Context context) { return mApplication; }

   @Provides
   @DatabaseInfo
   String provideDatabaseName() {
      return "skat_scores_app.db";
   }

   @Provides
   @DatabaseInfo
   Integer provideDatabaseVersion() {
      return 3;
   }
}

File: app/src/main/java/com/damhoe/skatscores/ApplicationModule.kt
package com.damhoe.skatscores

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

@InstallIn(SingletonComponent::class)
@Module
object ApplicationModule
{
    @Provides
    @DatabaseInfo
    fun provideDatabaseName() = "skat_scores_app.db"

    @Provides
    @DatabaseInfo
    fun provideDatabaseVersion() = 3
}
